### PR TITLE
Status updates and completion feature for the interactive list

### PR DESCRIPTION
**Pull request by jash-debug**
- Add status.js file to get status updates.
- Add event listener to the checkbox.
- Update items object's value for completed key upon user actions.
- Implement a function for the "Clear all completed" using (use filter() method).
- Store the updated array of items in local storage.
- fix linter errors
- Fix app behavior to make it more responsive
- make requested changes

### CODE REVIEW

File: src/functionality.js [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+  }
+
+  removeTask(index) {
+    this.tasksList.splice(index, 1);
+    for (let i = index; i < this.tasksList.length; i += 1) {
+      this.tasksList[i].index = i;
+    }
+    localStorage.setItem('tasks', JSON.stringify(this.tasksList));
+    this.updateDisplay();
+  }
```
Comment (levy002) [inline src/functionality.js:113]: - [ ] Kindly make sure the removing functionality works as expected. At this moment your app is not performing the delete functionality correctly. Kindly fix that.

   - ### Your website behavior

   chrome-capture-2022-10-12 (6)

**Review by levy002 — CHANGES_REQUESTED** [review]
## Hello @jash-debug   , :wave:
### I'm Levy, your PR reviewer

Good job so far :+1: 
There are some issues that you still need to work on to go to the next project but you are almost there!



## Status: REQUIRED CHANGES ♻️
_Check the comments under the review._

- [x] Kindly make your app behaves as expected because at this moment when the user clicks on the `checkbox` of any task, the first disappears and in order to get it back it requires the user to reload the page. Kindly fix that. 

 - ### Your app behavior
     chrome-capture-2022-10-12 (7)

- [x] Kindly make sure the task `index` start from `1` not from `0`. It means if we add the third task it must have an `index` of `3`.


## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @levy002 in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Qoosim — CHANGES_REQUESTED** [review]
#  *Status: Required Changes* :recycle: 

Hi @jash-debug, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights

-  No linter errors. :heavy_check_mark: 
-  Correct GitHub Flow. :heavy_check_mark: 
-  Good README file. :heavy_check_mark: 




 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

- [ ] Great work for implementing the suggestion by the previous reviewer. However, there's still little issue that need to be fixed. If I add one todo and click on the checked button, it gets disappeared as noticed by the previous reviewer and only reappears at page load. If you decided to add more todos and tried to check any of those, the first todo added gets disappeared. Kindly look into that and fix it. None of the added todo lists should get disappeared on checking any one of them, except deleted. Look at the screenshot below to confirm your app behaviour at this moment.

### Your App Behaviour


https://user-images.githubusercontent.com/26597432/201483061-be5a5924-b8d4-4420-9fe0-9947cdf17282.mp4

**Review by topeogunleye — APPROVED** [review]
### Hi @jash-debug, 

# Status: Approved ✔️
Excellent job on making the changes requested by the previous code reviewer. Keep it up 👏

Your project is complete! There is nothing else to say other than... it's time to merge it 💪
Congratulations! 🎉
congratulations
>
## To highlight :champagne: 
- All linters are passing :heavy_check_mark: 
- Good commit history :heavy_check_mark: 
- Followed Github flow :heavy_check_mark:
- Professional README file. :heavy_check_mark:
 
## [Optional] suggestions:
* Nothing to mention.
 
 _Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
 
Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @topeogunleye in your question so I can receive the notification.



------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**